middlewares/auth.middleware.js

```diff
@@ -1,19 +1,30 @@
 "use strict";
 const jwt = require("jsonwebtoken");
 
-const authMiddleware = async (req, res, next) => {
-
+exports.guard = async (req, res, next) => {
   try {
     let accessToken = req.cookies.jwt;
 
     if (!accessToken) {
       return res.status(403).send("Cannot access this route");
     }
-    const decode = jwt.verify(accessToken, process.env.ACCESS_TOKEN_SECRET);
-    next();
+    jwt.verify(accessToken, process.env.ACCESS_TOKEN_SECRET, (err, decoded) => {
+      if(err) {
+           return res.status(400).send('Session expired')
+      }
+      req.username = decoded.username;
+      next();
+      });
+    
```

Comment:
is it possible to write it in async/await?

Reply:
No, the npm document says that with a callback, this function is asynchronous.